pkg/convert/pot.go

```diff
@@ -110,47 +74,209 @@ func FromPot(b []byte) (model.Messages, error) {
 }
 
 func writeToPoTag(b *bytes.Buffer, tag poTag, str string) error {
-	encodedStr, err := json.Marshal(str) // use JSON encoding to escape special characters
-	if err != nil {
-		return fmt.Errorf("marshal %s: %w", tag, err)
+	switch tag { //nolint:exhaustive
+	case MsgStrPlural:
+		if err := writePluralForms(b, tag, str); err != nil {
+			return fmt.Errorf("write plural forms: %w", err)
+		}
+	default:
+		if err := writeDefault(b, tag, str); err != nil {
+			return fmt.Errorf("write default: %w", err)
+		}
 	}
 
-	encodedStr = encodedStr[1 : len(encodedStr)-1] // trim quotes
-	lines := strings.Split(string(encodedStr), "\\n")
+	return nil
+}
 
-	// Remove the empty string element. The line is empty when splitting by "\\n" and "\n" is the last character in str.
-	if lines[len(lines)-1] == "" {
-		lines = lines[:len(lines)-1]
-		lines[len(lines)-1] = lines[len(lines)-1] + "\\n" // add the "\n" back to the last line
+func writeDefault(b *bytes.Buffer, tag poTag, str string) error {
+	err, lines := getPoTagLines(str, tag)
+	if err != nil {
+		return fmt.Errorf("get po tag lines: %w", err)
 	}
 
 	if len(lines) == 1 {
-		_, err = fmt.Fprintf(b, "%s \"%s\"\n", tag, lines[0])
-		if err != nil {
+		if _, err = fmt.Fprintf(b, "%s \"%s\"\n", tag, lines[0]); err != nil {
 			return fmt.Errorf("write %s: %w", tag, err)
 		}
 
 		return nil
 	}
 
-	_, err = fmt.Fprintf(b, "%s \"\"\n", tag)
-	if err != nil {
+	if _, err = fmt.Fprintf(b, "%s \"\"\n", tag); err != nil {
 		return fmt.Errorf("write %s: %w", tag, err)
 	}
 
+	if err = writeMultiline(b, tag, lines); err != nil {
+		return fmt.Errorf("write multiline: %w", err)
+	}
+
+	return nil
+}
+
+func writePluralForms(b *bytes.Buffer, tag poTag, str string) error {
+	pluralForms, err := extractPluralForms(str)
+	if err != nil {
+		return fmt.Errorf("extract plural forms: %w", err)
+	}
+
+	for i := range pluralForms {
+		err, pluralLines := getPoTagLines(pluralForms[i], tag)
+		if err != nil {
+			return fmt.Errorf("get po tag lines: %w", err)
+		}
+
+		if len(pluralLines) == 1 {
```

Comment:
also think that here swithc would look more elegant that nested if
```go
switch len(pluralLines){
case 1:
...}
```

Reply:
Inside `for` loop I recommend `continue`, it keeps the flow more linear.

```go
if len(pluralLines) == 1 {
  // ...
  continue
}
```